users/views.py

```diff
@@ -51,5 +53,57 @@ def post(self,request):
                     return JsonResponse({'message': 'INVALID_CODE_NUMBER'}, status=401)
                 return JsonResponse({'message': 'CODE_SUCCESS'}, status=200)
                 
+        except KeyError:
+            return JsonResponse({'message': 'KEY_ERROR'}, status=401)
+
+class MobileSignUp(View):
+    def post(self,request):
+        try:
+            data     = json.loads(request.body)
+            mobile   = data['mobile']
+            password = data['password']     
+            birth    = data['birth']
+            gender   = data['gender']
+            nickname = data['nickname']
+           
+            if not re.match(r'^(?=.{8,16}$)(?=.*[a-z])(?=.*[0-9]).*$',password):
+                return JsonResponse({'message': 'INVALID_PASSWORD'}, status=401)
+            if User.objects.filter(mobile=mobile).exists():
+                return JsonResponse({'message' : 'EXIST_MOBILE'}, status=401)
+            if User.objects.filter(nickname=nickname).exists():
+                return JsonResponse({'message' : 'EXIST_NICKNAME'}, status=401)
+            new_library = Library.objects.create(name=f'{nickname}의 서재')
+            
+            encoded_pw = password.encode('utf-8')
+            hashed_pw  = bcrypt.hashpw(encoded_pw, bcrypt.gensalt())
+            encrypt_pw = hashed_pw.decode('utf-8')
```

Comment:
```suggestion
            encrypt_pw = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt())decode('utf-8')
```
줄일 수 있는 코드는 줄이도록 해주세요. 변수를 사용하는건 재사용성을 가정하고 작성하는게 좋습니다.

Reply:
넵 수정 진행하겠습니다.